src/ekat/ekat_workspace_impl.hpp

```diff
@@ -241,6 +260,36 @@ void WorkspaceManager<T, D>::Workspace::take_many_contiguous_unsafe(
   m_team.team_barrier();
 }
 
+template <typename T, typename D>
+template <typename S>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<typename WorkspaceManager<T, D>::template view_1d<S> >
+WorkspaceManager<T, D>::Workspace::take_n_size_block(
+  const char* name, const int n) const
+{
+#ifndef NDEBUG
+  change_num_used(n);
```

Comment:
I think we should keep the check that the data is contiguous here. This is similar to the check at the top of take_many_contiguous_unsafe.